pkg/controller/servicecatalog/provisionedproduct/setup.go

```diff
@@ -132,12 +131,31 @@ func (c *custom) lateInitialize(spec *svcapitypes.ProvisionedProductParameters,
 	return nil
 }
 
-func (c *custom) isUpToDate(_ context.Context, ds *svcapitypes.ProvisionedProduct, resp *svcsdk.DescribeProvisionedProductOutput) (bool, string, error) {
+func preCreate(_ context.Context, cr *svcapitypes.ProvisionedProduct, input *svcsdk.ProvisionProductInput) error {
+	input.ProvisionToken = aws.String(genIdempotencyToken())
+	if cr.GetName() != meta.GetExternalName(cr) {
+		return errors.New(errCreatExternalNameIsNotValid)
+	}
+	input.ProvisionedProductName = aws.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func (c *custom) preUpdate(_ context.Context, cr *svcapitypes.ProvisionedProduct, input *svcsdk.UpdateProvisionedProductInput) error {
+	input.UpdateToken = aws.String(genIdempotencyToken())
+	if cr.GetName() == meta.GetExternalName(cr) {
+		input.ProvisionedProductName = aws.String(meta.GetExternalName(cr))
+	} else {
+		input.ProvisionedProductId = aws.String(meta.GetExternalName(cr))
+	}
+	return nil
+}
+
+func (c *custom) isUpToDate(ctx context.Context, ds *svcapitypes.ProvisionedProduct, resp *svcsdk.DescribeProvisionedProductOutput) (bool, string, error) { // nolint:gocyclo
 	// If the product is undergoing change, we want to assume that it is not up-to-date. This will force this resource
 	// to be queued for an update (which will be skipped due to UNDER_CHANGE), and once that update fails, we will
 	// recheck the status again. This will allow us to quickly transition from UNDER_CHANGE to AVAILABLE without having
 	// to wait for the entire polling interval to pass before re-checking the status.
-	if pointer.StringDeref(ds.Status.AtProvider.Status, "") == string(svcapitypes.ProvisionedProductStatus_SDK_UNDER_CHANGE) {
```

Comment:
ds doesn't contain anything in ds.Status.AtProvider, it will be be set later. that is why I replace it to resp.ProvisionedProductDetail.Status